### PR TITLE
Add comment table and bean.

### PR DESCRIPTION
**Pull request by IkramMaalej**

### CODE REVIEW

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/CommentDAO.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        return retrieveAllObjects(Comment.class);
+    }
+
+    @Override
+    public List<Comment> getAll(int offset, int size) throws DAOException {
+        return retrieveObjects("FROM Comment ORDER BY id ASC", offset, size);
+    }
+
+    @Override
+    public List<Comment> getAllNotIndexed(int offset, int size) throws DAOException {
```
Comment (Beacze) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/CommentDAO.java]: This method shouldn't be implemented - comment is not indexed bean.

File: Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                    instance = new CommentService();
+                }
+            }
+        }
+        return instance;
+    }
+
+    @Override
+    public List<Comment> getAllForSelectedClient() {
+        return null;
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java]: Please return empty array, not null.

**Comment by Beacze** [comment]
The warnings in CommentDAO can be ignored, only not used field should be removed.

File: Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import org.primefaces.model.SortOrder;
+
+public class CommentService extends SearchDatabaseService<Comment, CommentDAO> {
+
+    private static CommentService instance = null;
+
+    /**
+     * Constructor necessary to use searcher in child classes.
+     */
+    public CommentService() {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java]: Constructor needs to be private

File: Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    /**
+     * Return singleton variable of type TaskService.
+     *
+     * @return unique instance of TaskService
+     */
+    public static CommentService getInstance() {
+        if (Objects.equals(instance, null)) {
+            synchronized (TaskService.class) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java]: Wrong class (TaskService?)

**Comment by IkramMaalej** [comment]
@Kathrin-Huber This Codacy warning is not correct in this case.

**Pull request merged**